FirebaseMLModelDownloader/Sources/ModelInfoRetriever.swift

```diff
@@ -209,15 +213,49 @@ extension ModelInfoRetriever {
     }
     // TODO: Possibly improve handling invalid server responses.
     guard let downloadURL = URL(string: modelInfoJSON.downloadURL) else {
-      throw DownloadError.internalError(description: "Invalid model download URL from server.")
+      throw DownloadError
+        .internalError(description: ModelInfoRetriever.invalidModelDownloadURLErrorDescription)
     }
     let modelHash = modelHash
     let size = Int(modelInfoJSON.size) ?? 0
+    let dateFormatter = ModelInfoRetriever.getDateFormatter()
+    guard let expiryTime = dateFormatter.date(from: modelInfoJSON.urlExpiryTime) else {
+      throw DownloadError
+        .internalError(description: ModelInfoRetriever
+          .invalidModelDownloadURLExpiryTimeErrorDescription)
+    }
     return RemoteModelInfo(
       name: modelName,
       downloadURL: downloadURL,
       modelHash: modelHash,
-      size: size
+      size: size,
+      urlExpiryTime: expiryTime
     )
   }
 }
+
+/// Possible error messages for model info retrieval.
+extension ModelInfoRetriever {
+  /// Error descriptions.
+  private static let tokenErrorDescription = "Error retrieving FIS token."
```

Comment:
optional: It may be a bit nicer to read if we wrap the descriptions into an `enum` for name spacing and remove `Description` part from the constant names, e.g.:
```
extension ModelInfoRetriever {
  private enum Description {
    static let tokenError = "Error retrieving FIS token."
    ...
  }
}
```

Though I don't have a strong opinion here.

Reply:
+1, although likely named `ErrorDescription`

Reply:
Good idea, done.